### PR TITLE
Add test for https://github.com/resteasy/Resteasy/pull/2578

### PR DESCRIPTION
**Pull request by jimma**
This PR is rebased with https://github.com/resteasy/Resteasy/pull/2578 and changes  the following things: 
1. Add executor to SSEPublisher to avoid block the Publisher emit
2. Add system property "resteasy.microprofile.sseclient.buffersize" to configure the Subscription queue size.
3. Add test for https://github.com/resteasy/Resteasy/pull/2578 and configure the Subscription queue size to 5 to check the overflow behavior.

### CODE REVIEW

**Comment by jimma** [comment]
@cescoffier  I added one test for SSEPublisher to check the overflow behavior for https://github.com/resteasy/Resteasy/pull/2578, please review.

File: resteasy-client-microprofile-base/src/main/java/org/jboss/resteasy/microprofile/client/RestClientBuilderImpl.java [hunk 1]
```diff
@@ -20,6 +20,8 @@
 import org.jboss.resteasy.microprofile.client.header.ClientHeadersRequestFilter;
 import org.jboss.resteasy.microprofile.client.impl.MpClient;
 import org.jboss.resteasy.microprofile.client.impl.MpClientBuilderImpl;
+import org.jboss.resteasy.microprofile.client.publisher.MpPublisherMessageBodyReader;
+//import org.jboss.resteasy.microprofile.client.publisher.MpPublisherMessageBodyReader;
```
Comment (cescoffier) [inline resteasy-client-microprofile-base/src/main/java/org/jboss/resteasy/microprofile/client/RestClientBuilderImpl.java:24]: Shouldn't we remove this commented line?

File: resteasy-client-microprofile-base/src/main/java/org/jboss/resteasy/microprofile/client/publisher/SSEPublisher.java [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+     * Reads the server-sent event from the {@code input} and pass them to the processor.
+     * The processor handles the downstream requests and buffer/drop items according to them.
+     *
+     * @param processor the stream
+     * @param input     the SSE input
+     */
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    private void pump(final SSEProcessor processor, final SseEventInputImpl input) {
+        Map<Class<?>, Object> contextDataMap = ResteasyContext.getContextDataMap();
+        executor.execute(() -> {
```
Comment (cescoffier) [inline resteasy-client-microprofile-base/src/main/java/org/jboss/resteasy/microprofile/client/publisher/SSEPublisher.java:66]: should it handle rejection?

**Review by cescoffier — CHANGES_REQUESTED** [review]
LGTM. I've made 2 minor comments.

File: resteasy-client-microprofile-base/src/main/java/org/jboss/resteasy/microprofile/client/RestClientBuilderImpl.java (hunk 1 above)
Comment (jimma) [reply resteasy-client-microprofile-base/src/main/java/org/jboss/resteasy/microprofile/client/RestClientBuilderImpl.java:24]: Thanks. Fixed.

File: resteasy-client-microprofile-base/src/main/java/org/jboss/resteasy/microprofile/client/publisher/SSEPublisher.java (hunk 2 above)
Comment (jimma) [reply resteasy-client-microprofile-base/src/main/java/org/jboss/resteasy/microprofile/client/publisher/SSEPublisher.java:66]: Do you mean handle executor injection ?

Comment (cescoffier) [reply resteasy-client-microprofile-base/src/main/java/org/jboss/resteasy/microprofile/client/publisher/SSEPublisher.java:66]: if the executor is stopped (for whatever reason), you cannot submit another task. You will receive a `RejectedExecutionException`. What should we do in this case? Log and return a failed publisher? Log and return a `Publisher` just emitting the completion signal?

Comment (jimma) [reply resteasy-client-microprofile-base/src/main/java/org/jboss/resteasy/microprofile/client/publisher/SSEPublisher.java]: @cescoffier Thanks for pointing out this issue.  I modified to this to create a new thread to run this if there is RejectedExecutionException thrown.

**Review by cescoffier — APPROVED** [review]

**Review by asoldano — APPROVED** [review]

**Pull request merged**